pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/PulsarMessage.java

```diff
@@ -1095,7 +1099,9 @@ void acknowledgeInternal() throws JMSException {
       return;
     }
     try {
-      consumer.acknowledge(receivedPulsarMessage, this, pulsarConsumer);
+      if (consumer != null) {
+        consumer.acknowledge(receivedPulsarMessage, this, pulsarConsumer);
```

Comment:
in this case we should throw an IllegalStateException here.

how did you  see an error here ?

Reply:
I suspected that an IllegalStateException was the better approach, so I can make that change. As for how I ended up here, I am still trying to figure that out, but I suspect it has something to do with not closing the PulsarSession and/or PulsarConnection object between calls.